README.md

```diff
@@ -40,9 +40,11 @@ For complete API documentation, see the [Ably documentation](https://www.ably.io
 and require as:
 
 ```javascript
-var Ably = require('ably')
+var Ably = require('ably');
 ```
 
+For the version of the library where async methods return promises, use `var Ably = require('ably/promises');` instead. (You can also use `require('ably/callbacks')` to get a version that for now behaves the same as the normal lib, but is guaranteed to not return promises even if we make returning a promise the default for `require('ably')` in some future major version).
```

Comment:
Personally I think I would have a new section, just after the "Supported Platforms" section saying something like the suggested text below. This makes things clearer and avoids repetition in the following sections.

## Async API style

This library exposes two API variants: one, the original (and presently the default) callback-based API. This follows the usual node.js error-first callback style; and second, a promises-based API. The API in use is selected explicitly by requiring that specific variant when including the library. The usage instructions below make reference to both variants.

At this library version, and for all future 1.x versions, the callback-based API is the default. However, a move to adopt the promise-based API as the default is possible at the next major release; therefore, it is recommended that calling code using callbacks migrates to explicitly requiring the callback based API, as highlighted in the usage instructions below.

Reply:
Added (though tweaked the wording to make clearer that even in the promise variant you can pass a callback, and the compatibility break is more to do with returning a promise when no callback is passed)